lib/cli/src/backend.rs

```diff
@@ -844,116 +665,63 @@ impl BackendType {
 
         // Check if the backend supports all required features
         if !supported.contains_features(required_features) {
             tracing::info!("Backend {:?} doesn't support all required features", self);
             tracing::info!("Supported: {:?}", supported);
             tracing::info!("Required: {:?}", required_features);
-            return Ok(false);
+            return false;
         }
 
-        // Try creating an engine to verify features work
-        let engine = self.get_engine(&target, required_features)?;
-
-        // If we got this far, basic feature requirements are met
-        Ok(true)
-    }
-
-    /// Filter enabled backends based on required WebAssembly features
-    #[cfg(feature = "compiler")]
-    pub fn filter_by_features(required_features: &Features) -> Vec<Self> {
-        let available_backends = Self::enabled();
-        let target = wasmer_compiler::types::target::Target::default();
-
-        // Special handling for exceptions - if exceptions are required, make sure LLVM is included
-        if required_features.exceptions {
-            #[cfg(feature = "llvm")]
-            {
-                tracing::info!("Module requires exceptions, forcing inclusion of LLVM backend");
-
-                // If LLVM is available, return it as the only option for exceptions
-                for backend in &available_backends {
-                    if matches!(backend, Self::LLVM) {
-                        return vec![Self::LLVM];
-                    }
-                }
-            }
-        }
-
-        // Normal filtering for other features
-        available_backends
-            .into_iter()
-            .filter(|backend| {
-                // First check using the Engine::supported_features_for_backend
-                // which queries the actual engine implementations
-                match wasmer::Engine::supported_features_for_backend(&backend.into(), &target) {
-                    supported if supported.contains_features(required_features) => true,
-                    _ => {
-                        // Fallback to our hardcoded checks if the engine-based check fails
-                        match backend.supports_features(required_features) {
-                            Ok(true) => true,
-                            _ => false,
-                        }
-                    }
-                }
-            })
-            .collect()
+        true
     }
 }
 
 impl From<&BackendType> for wasmer::BackendKind {
     fn from(backend_type: &BackendType) -> Self {
         match backend_type {
-            BackendType::Singlepass => wasmer::BackendKind::Sys,
-            BackendType::Cranelift => wasmer::BackendKind::Sys,
-            BackendType::LLVM => wasmer::BackendKind::Sys,
+            BackendType::Singlepass => wasmer::BackendKind::Singlepass,
+            BackendType::Cranelift => wasmer::BackendKind::Cranelift,
+            BackendType::LLVM => wasmer::BackendKind::LLVM,
```

Comment:
When building with LLVM disable,
```
error[E0599]: no variant or associated item named `LLVM` found for enum `BackendKind` in the current scope
   --> lib/cli/src/backend.rs:708:55
    |
708 |             BackendType::LLVM => wasmer::BackendKind::LLVM,
    |                                                       ^^^^ variant or associated item not found in `BackendKind`
```

Should we fallback to Cranelift in this case or headless? @syrusakbary  @xdoardo

Reply:
I checked the code, this is now solved